### PR TITLE
feat(add-get-single-file): update with get single file

### PR DESCRIPTION
**Pull request by adiatma**
# Description

Add get single file fitur.

## Usecase

Example in svg directories, we have more file svg, but i want to update a single file svg, with this fitur we can convert a single file svg.

Screen Shot 2021-07-26 at 13 44 10

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -16,57 +20,79 @@ async function getFiles(dir: string): Promise<string[]> {
...
     .concat(...files)
     .filter((file) => path.extname(file) === ".svg");
 }
 
+const parseToJSXComponent = (
+  filePath: string,
+  outputPath: string,
+  callback?: () => any
+) => {
+  const spinner = ora("generating components").start();
```
Comment (adeyahya) [inline src/index.ts]: this will cause multiple spinner, this should be called inside command body function and updated every iteration in iterate function

**Review by adeyahya — CHANGES_REQUESTED** [review]
please consider my comment

File: src/index.ts [hunk 2]
```diff
@@ -16,57 +20,79 @@ async function getFiles(dir: string): Promise<string[]> {
...
+    );
+  });
+};
+
 const program = new Command();
 program.argument("<sourceDir>").requiredOption("-o --output <value>");
 
 program.action(async (sourcePath: string, { output }) => {
-  const spinner = ora("generating components").start();
+  const stats = fs.lstatSync(sourcePath);
```
Comment (adeyahya) [inline src/index.ts:71]: you should initiate spinner in here

File: src/index.ts [hunk 3]
```diff
@@ -16,57 +20,79 @@ async function getFiles(dir: string): Promise<string[]> {
...
-            spinner.text = `${componentName + ".tsx"} generated`;
-          }
-          iterate();
-        }
-      );
-    });
-  };
+    const iterate = () => {
+      const filePath = filesIterator.next().value;
+      parseToJSXComponent(filePath, outputPath, iterate);
```
Comment (adeyahya) [inline src/index.ts:89]: and update spinner in here `spinner.text = `${filePath} converted successfully`;`

File: src/index.ts (hunk 1 above)
Comment (adiatma) [reply src/index.ts]: Updated, check again pak @adeyahya

**Review by adeyahya — APPROVED** [review]

**Pull request merged**